File: flutter_hoo/lib/db/shoe.dart
class Shoe {
  final int id;
  final String name;
  final String description;
  final double price;
  final String brand;
  final String imageUrl;

  Shoe(this.name, this.description, this.price, this.brand,
      this.imageUrl,{this.id});

  factory Shoe.fromJson(Map<String,dynamic> map){
    return Shoe(map['name'], map['description'], map['price'], map['brand'], map['imageUrl'],id: map['id']);
  }

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'price': price,
      'brand': brand,
      'imageUrl': imageUrl
    };
  }
}

File: compose_hoo/app/src/main/java/com/example/composehoo/db/data/Shoe.kt
package com.example.composehoo.db.data

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

/**
 * 鞋表
 */
@Entity(tableName = "shoe")
data class Shoe(
    @ColumnInfo(name = "shoe_name") var name: String
    , @ColumnInfo(name = "shoe_description") val description: String
    , @ColumnInfo(name = "shoe_price") val price: Float
    , @ColumnInfo(name = "shoe_brand") val brand: String
    , @ColumnInfo(name = "shoe_imgUrl") val imageUrl: String
) {
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id")
    var id: Long = 0

    fun getPriceStr():String{
        return price.toString()
    }

    override fun toString(): String {
        return "id: ${id}, shoe_name: ${name}, description: ${description}, shoe_price: ${price}, brand: ${brand}"
    }
}